src/ButtonPage.js

```diff
@@ -1,10 +1,121 @@
 import React, { Component } from 'react';
+import ComponentPage from './ComponentPage.js';
 import HeaderBar from './HeaderBar.js';
+import {MDCRipple} from '@material/ripple';
+
+import './ButtonPage.scss';
 
 class ButtonPage extends Component {
   render() {
     return (
-      <HeaderBar title='Button'/>
+      <div>
+        <HeaderBar title='Button'/>
+        <ComponentPage
+          hero={<ButtonHero/>}
+          sidebarComponentSelector='.sidebar-buttons'
+          title='Buttons'
+          description='Buttons communicate an action a user can take. They are typically placed throughout your UI, in places like dialogs, forms, cards, and toolbars.'
+          designLink='https://material.io/guidelines/components/buttons.html'
+          docsLink='https://material.io/components/web/catalog/buttons/'
+          sourceLink='https://github.com/material-components/material-components-web/tree/master/packages/mdc-button'
+          demos={<ButtonDemos/>}
+        />
+      </div>
+    );
+  }
+}
+
+class ButtonHero extends Component {
+  componentDidMount() {
+    const buttons = document.querySelectorAll('.mdc-button');
+    buttons.forEach(button => new MDCRipple(button));
+  }
+
+  render() {
+    return (
+      <div>
+        <button className='mdc-button'>
+          Text
+        </button>
+        <button className='mdc-button mdc-button--raised'>
+          Raised
+        </button>
+        <button className='mdc-button mdc-button--unelevated'>
+          Unelevated
+        </button>
+        <button className='mdc-button mdc-button--stroked'>
+          Stroked
+        </button>
+      </div>
+    );
+  }
+}
+
+class ButtonDemos extends Component {
+  componentDidMount() {
+    const buttons = document.querySelectorAll('.mdc-button');
+    buttons.forEach(button => new MDCRipple(button));
+  }
+
+  render() {
+    return (
+      <div>
+        <h3 className='mdc-typography--subheading2'>Text Button</h3>
```

Comment:
Sounds good! I broke ButtonDemos into a render function for each variant like you suggested but I left the `<button>` HTML as is because I didn't want to wrap it to the point where it'll just be the MDC React button

Reply:
Oh are you saying we'll eventually swap this out with MDC React Button?

Reply:
Possibly, but I mean that for now since most of the components will be vanilla components, let's just make all of them vanilla components.
Whether or not to swap them out with React components will be a separate discussion.